### PR TITLE
Clock sync

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: src/clocksync/clocksync.hpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    ClockSyncSenderItf*          mSender {};
+    aos::ConnectionPublisherItf* mConnectionPublisher {};
+    aos::Thread<>                mThreadClockSync = {};
+    aos::Mutex                   mMutex;
+    bool                         mFinish {};
+    bool                         mSendClockSyncRequest {};
+    aos::ConditionalVariable     mClockSyncRequestCV {mMutex};
+};
+
+#endif
```
Comment (al1img) [inline src/clocksync/clocksync.hpp:107]: New line at the end.

File: src/clocksync/log.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#define LOG_HPP_
+
+#include "logger/logger.hpp"
+
+#define LOG_DBG() LOG_MODULE_DBG(static_cast<aos::LogModule::EnumType>(Logger::Module::eClockSync))
+#define LOG_INF() LOG_MODULE_INF(static_cast<aos::LogModule::EnumType>(Logger::Module::eClockSync))
+#define LOG_WRN() LOG_MODULE_WRN(static_cast<aos::LogModule::EnumType>(Logger::Module::eClockSync))
+#define LOG_ERR() LOG_MODULE_ERR(static_cast<aos::LogModule::EnumType>(Logger::Module::eClockSync))
+
+#endif
```
Comment (al1img) [inline src/clocksync/log.hpp:18]: ditto

File: tests/clocksync/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+target_include_directories(app PRIVATE ${APPLICATION_SOURCE_DIR}/../../src)
+target_include_directories(app PRIVATE ${APPLICATION_SOURCE_DIR}/../../../aos_core_lib_cpp/include)
+target_include_directories(app PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+
+# ######################################################################################################################
+# Target
+# ######################################################################################################################
+
+target_sources(app PRIVATE src/main.cpp ../../src/clocksync/clocksync.cpp)
```
Comment (al1img) [inline tests/clocksync/CMakeLists.txt:23]: ditto

File: src/clocksync/clocksync.hpp (hunk 1 above)
Comment (MykolaSolyanko) [reply src/clocksync/clocksync.hpp]: Done

File: src/clocksync/log.hpp (hunk 2 above)
Comment (MykolaSolyanko) [reply src/clocksync/log.hpp]: Done

File: tests/clocksync/CMakeLists.txt (hunk 3 above)
Comment (MykolaSolyanko) [reply tests/clocksync/CMakeLists.txt]: Done

File: src/clocksync/log.hpp (hunk 2 above)
Comment (al1img) [inline src/clocksync/log.hpp]: New Line at the end

**Pull request closed without merging**